core/common/src/main/java/alluxio/worker/block/io/BlockReader.java

```diff
@@ -46,4 +48,18 @@
    * @return channel
    */
   ReadableByteChannel getChannel();
+
+  /**
+   * Transfers data from this reader to the buffer.
+   *
+   * @param buf the byte buffer
+   * @return the number of bytes transferred, -1 if the end of the block is reached
+   * @throws IOException if any I/O errors occur
+   */
+  int transferTo(ByteBuf buf) throws IOException;
```

Comment:
it seems for this `transferTo` API it should be fine to always use `transferTo(ByteBuffer buf)`, right?  because here buf is the output and we could always create a `ByteBuf` from buf of type `ByteBuffer` in caller.

Is the problem in `transferFrom` API? I found if we are tranferring from readable bytes, it should be able to create nio ByteBuffer from [nioBuffers()](https://netty.io/4.0/api/io/netty/buffer/ByteBuf.html#nioBuffers())